### PR TITLE
Support path-only when referencing ssl certificate in compute_target_https_proxy

### PR DESCRIPTION
**Pull request by rosbo**
issue #47 

* Added support for both `full url` and `path-only` when referencing ssl certificate in * compute_target_https_proxy. This means both `https://www.googleapis.com/compute/v1/projects/MY-PROJECT/global/sslCertificates/MY_CERT` and `projects/MY-PROJECT/global/sslCertificates/MY_CERT` are valid.
* Simplified the read/update logic for ssl certificate since a proxy must have exactly one ssl certificate.
* Added more assertions to acceptance test
* Added schema validation of ssl certificate field

cc/ @danawillow @rileykarson

### CODE REVIEW

File: google/resource_compute_target_https_proxy.go [hunk 1]
```diff
@@ -256,3 +212,14 @@ func resourceComputeTargetHttpsProxyDelete(d *schema.ResourceData, meta interfac
 	d.SetId("")
 	return nil
 }
+
+func toCanonicalSslCertificate(v interface{}) string {
+	value := v.(string)
+
+	m := regexp.MustCompile(sslCertificateRegex).FindStringSubmatch(value)
+	if m == nil {
```
Comment (rosbo) [inline google/resource_compute_target_https_proxy.go]: It should never be nil. Removed this line.

File: google/resource_compute_target_https_proxy_test.go [hunk 2]
```diff
@@ -7,52 +7,83 @@ import (
...
 }
 
+func TestAccComputeTargetHttpsProxy_invalidCertificate(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeTargetHttpsProxyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: `
```
Comment (rosbo) [inline google/resource_compute_target_https_proxy_test.go]: Done

File: google/resource_compute_target_https_proxy.go [hunk 3]
```diff
@@ -26,7 +32,13 @@ func resourceComputeTargetHttpsProxy() *schema.Resource {
 			"ssl_certificates": &schema.Schema{
 				Type:     schema.TypeList,
 				Required: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateRegexp(sslCertificateRegex),
+					StateFunc:    toCanonicalSslCertificate,
+				},
+				MinItems: 1,
```
Comment (rosbo) [inline google/resource_compute_target_https_proxy.go]: fixed

**Comment by danawillow** [comment]
Looks good, thanks! Merge at will.

**Pull request merged**